### DISCUSSION TITLE
why struct no convertible to nullable<struct>?!

### DISCUSSION THREAD

**Discussion by cw-1988** [dotnet/csharplang#5933] [Language Ideas]
it would be great if those primitive types int long etc could have the IConvertible<Nullable[T]> So i can use generics to cast a struct onto its nullable version. It would also be great for other use cases where object can ACT AS the generic parameter because they are convertible to it, while not having to inheriting it.

**Comment by jnm2** [comment]
I don't think there's a generic `IConvertible<T>` interface. If there was, what would it accomplish which directly casting to `T?` could not accomplish?

**Reply by cw-1988** [reply]
it would help because T? is a non-constraint meaning the ? does nothing right now. Nullable<T> also isn't allowed as a constraint. IConvertible<T> would be icing on the cake and enver implemented because of its opportunity cost being too high as Eric Lippert would call it... but nullable actually being able to be used as a generic constraint, convertible to T and being detectable in reflection would make it's addition to the c# code base actually complete...

**Reply by huoyaoyuan** [reply]
> convertible to T and being detectable in reflection

Nullable structs aren't really "detectable" in reflection. They are boxed into the same form of underlying struct, and can be directly converted using box and unbox.

**Reply by jnm2** [reply]
@huoyaoyuan's point is why I'm wondering what concrete scenario is being considered here.
